### PR TITLE
Pxapi2/feature/search

### PR DESCRIPTION
**Pull request by MikaelNordberg**
Functionality to create and update search index

### CODE REVIEW

File: Px.Search/TableInformation.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Px.Search
+{
+    public class TableInformation
```
[the remark this answers is not in the document]
Comment (MikaelNordberg) [reply Px.Search/TableInformation.cs:9]: Constructor has been added

File: Px.Search.Lucene/LuceneBackend.cs [hunk 2]
```diff
@@ -8,18 +10,56 @@ namespace Px.Search.Lucene
...
+            string path = GetIndexDirectoryPath();
+            return new LuceneSearcher(path);
+        }
+
+ 
+        /// <summary>
+        /// Get path to the specified index directory 
+        /// </summary>
+        /// <returns>Physical path to Lucene index directory</returns>
+        private string GetIndexDirectoryPath()
```
[the remark this answers is not in the document]
Comment (MikaelNordberg) [reply Px.Search.Lucene/LuceneBackend.cs]: Method has been moved

File: Px.Search/TableInformation.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MikaelNordberg) [reply Px.Search/TableInformation.cs:9]: Constructor has been added

File: PxWeb/Code/Api2/DataSource/Cnmm/CnmmDataSource.cs [hunk 3]
```diff
@@ -56,12 +57,15 @@ public PxMenuBase CreateMenu(string id, string language, out bool selectionExist
                             if (item is TableLink)
                             {
                                 TableLink tbl = (TableLink) item;
-                                string tblId = tbl.ID.Selection;
+                                //string tblId = tbl.ID.Selection;
```
[the remark this answers is not in the document]
Comment (MikaelNordberg) [reply PxWeb/Code/Api2/DataSource/Cnmm/CnmmDataSource.cs]: Comment has been removed

**Review by likp — APPROVED** [review]

**Pull request merged**